### PR TITLE
[TSK-160] add Transaction page and connection server

### PR DESCRIPTION
**Pull request by Hoocki**
to check the pagination of transactions, you need to turn on the server (since the client does not know how many transactions exist, so pagination is impossible without a server)

### CODE REVIEW

File: frontend/src/common/constansts/content.ts [hunk 1]
```diff
@@ -10,4 +10,11 @@ export const Content = {
     TERMS: "I agree with terms, conditions, and privacy policy",
     ACCOUNT_EXIST: "Already have an account? ",
     ACCOUNT_NOT_EXIST: "Don't have an account? "
+}
+
+export const TableContent = {
```
Comment (adsukharev) [inline frontend/src/common/constansts/content.ts]: move it to table component

File: frontend/src/pages/main/transactions/components/tableTransactions/TableTransactions.tsx [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+                rowsPerPageOptions={[pageTransactions.size]}
+                component="div"
+                slotProps={{
+                    actions: {
+                        nextButton: {
+                            disabled: isLastPage
+                        }
+                    }
+                }}
+                labelDisplayedRows={() => {
```
Comment (adsukharev) [inline frontend/src/pages/main/transactions/components/tableTransactions/TableTransactions.tsx]:
```suggestion
                labelDisplayedRows={() => `Current page of transactions: ${pageTransactions.page}`}

```

File: frontend/src/pages/main/transactions/components/tableTransactions/TableTransactions.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+                    return `Current page of transactions: ${pageTransactions.page}`;
+                }}
+                count={-1}
```
Comment (adsukharev) [inline frontend/src/pages/main/transactions/components/tableTransactions/TableTransactions.tsx:99]: what is -1?

File: frontend/src/pages/main/transactions/components/tableTransactions/props.ts [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import {IWallet} from "../../../../../types/wallet";
+
+export interface TableTransactionsProps {
+    wallet: IWallet
```
Comment (adsukharev) [inline frontend/src/pages/main/transactions/components/tableTransactions/props.ts]: pass only what you need

File: frontend/src/common/constansts/time.ts [hunk 5]
```diff
@@ -1,4 +1,13 @@
...
-}
\ No newline at end of file
+}
+
+export const MONTHS = [
+    "January", "February", "March", "April", "May", "June",
+    "July", "August", "September", "October", "November", "December"
+];
+
+export const YEARS = Array.from({ length: 5 }, (_, i) => 2024 - i);
```
Comment (adsukharev) [inline frontend/src/common/constansts/time.ts]: Are you going to create PR each year to increase current year by one?

Few lines above you have Time. CURRENT_YEAR

File: frontend/src/pages/main/transactions/components/generatePdf/GeneratePdf.tsx [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+            })
+        } else {
+            setEndPeriod({...endPeriod, year});
+            setReport({
+                ...report,
+                periodEnd: new Date(year, endPeriod.month, endPeriod.day)
+            })
+        }
+        updateDays(startPeriod.month, year, periodType);
+    };
```
Comment (adsukharev) [inline frontend/src/pages/main/transactions/components/generatePdf/GeneratePdf.tsx]: You should already see duplication code. handleDaysChange, handleMonthChange, handleYearChange are perfect examples

File: frontend/src/pages/main/transactions/components/generatePdf/GeneratePdf.tsx [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+    const handleGenerate = async () => {
+        if (report.periodStart.getTime() > report.periodEnd.getTime()) {
+            dispatch(showNotification({show: true, type: NotificationType.ERROR, message: Notifications.PERIOD_TIME}));
+            return;
+        }
+        dispatch(showNotification({
+            show: true,
+            type: NotificationType.SUCCESS,
+            message: Notifications.SEND_GENERATE_PDF
+        }));
```
Comment (adsukharev) [inline frontend/src/pages/main/transactions/components/generatePdf/GeneratePdf.tsx:117]: You have to show successful notification , after request is successful.
Just move it to the next line after ` await sendReport(report);`

File: frontend/src/pages/main/transactions/components/generatePdf/GeneratePdf.tsx [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+        await sendReport(report);
+    }
+
+    useEffect(() => {
+        updateReportUser()
```
Comment (adsukharev) [inline frontend/src/pages/main/transactions/components/generatePdf/GeneratePdf.tsx:121]: not sure why you need it.  Why cannot you set it `{...report, userId: loggedUser.id}` as default values?

File: frontend/src/pages/main/transactions/components/generatePdf/GeneratePdf.tsx [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+    }, []);
+
+    return (
+        <Card className="card-background card-payment card-generate-pdf">
+            <CardContent className="card-content">
+                <Box className="balance-info">
+                    <CardTitle title={"Generate PDF report"}/>
```
Comment (adsukharev) [inline frontend/src/pages/main/transactions/components/generatePdf/GeneratePdf.tsx]: move it to constant

File: frontend/src/pages/main/transactions/components/generatePdf/GeneratePdf.tsx [hunk 10]
```diff
@@ -0,0 +1,202 @@
...
+                    <Box className="report-date report-end-date">
```
Comment (adsukharev) [inline frontend/src/pages/main/transactions/components/generatePdf/GeneratePdf.tsx]: Looks like you can create a separate component. Then reuse it for `From` and `To` forms

**Review by adsukharev — APPROVED** [review]
Don't want to write lots of comments. Just finish it and go to the next task.
P.s don't forget to follow branch naming pattern

File: frontend/src/pages/main/transactions/components/tableTransactions/TableTransactions.tsx (hunk 3 above)
Comment (Hoocki) [reply frontend/src/pages/main/transactions/components/tableTransactions/TableTransactions.tsx:99]: The total number of rows.
To enable server side pagination for an unknown number of items, provide -1.

File: frontend/src/pages/main/transactions/components/generatePdf/GeneratePdf.tsx (hunk 8 above)
Comment (Hoocki) [reply frontend/src/pages/main/transactions/components/generatePdf/GeneratePdf.tsx:121]: userId does not have time to be embedded in the report if it is inserted by default

**Pull request merged**